great_expectations/rule_based_profiler/domain.py

```diff
@@ -211,17 +211,36 @@ def to_json_dict(self) -> dict:
 def convert_dictionary_to_domain_kwargs(
     source: Union[T, dict]
 ) -> Union[T, DomainKwargs]:
-    if not isinstance(source, dict):
-        return source
+    if isinstance(source, dict):
+        return _convert_dictionary_to_domain_kwargs(source=DomainKwargs(source))
 
-    return _convert_dictionary_to_domain_kwargs(source)
+    # Must allow for non-dictionary source types, since their internal nested structures may contain dictionaries.
```

Comment:
Same comment regarding the method naming

Reply:
@NathanFarmer Yes, same as before -- there are the total of three such things.  Thanks.